subsys/bluetooth/mesh/Kconfig

```diff
@@ -9,6 +9,7 @@ menuconfig BT_MESH
 	select TINYCRYPT_AES
 	select TINYCRYPT_AES_CMAC
 	select BT_HOST_CCM
+	select BT_EXT_ADV
```

Comment:
I was first thinking about doing it the way you suggested, but I was trying to avoid changing anything in bluetooth. It is still possible to use BT_MESH_ADV_LEGACY, and turn off BT_CTRL_ADV_EXT, but it will not be possible to turn off BT_EXT_ADV. But I think you are correct and we need to change it this way.

Reply:
`default y if BT_MESH`:
Will this generate compilation warning if someone were to set `BT_EXT_ADV=n` in their prj.conf?

Reply:
It will not give a warning when BT_EXT_ADV=n is set.